discoveryapis_generator/lib/src/dart_resources.dart

```diff
@@ -670,7 +670,7 @@ DartResourceMethod _parseMethod(
 
   // Check if we have a request object, if so parse it's type.
   MethodParameter? dartRequestParameter;
-  if (method.request != null) {
```

Comment:
Can you explain the diff in the generator?

Reply:
I was follow the contrib guide and ran the generator but it crashed out with v1/v2 json files in play. I decided to debug it expecting I'd nee to contact you to create some diff/patches. But debugging and stepping it I found the code would pass in nullable references but always treat them as non-null. 

I've also update those type declarations to make this clear.